src/CodeGenerator/src/Definition/Operation.php

```diff
@@ -176,4 +176,28 @@ public function isDeprecated(): bool
     {
         return $this->data['deprecated'] ?? false;
     }
+
+    private function getInputShape(): ?Shape
+    {
+        if (isset($this->data['input']['shape'])) {
+            return ($this->shapeLocator)($this->data['input']['shape']);
+        }
+
+        $output = $this->getOutput();
+
+        if (null !== $output) {
+            return Shape::create(
+                str_replace('Output', 'Request', $output->getName()),
```

Comment:
Sometime the suffix is `Response` (in AppSync for instance), sometimes `Result` (in CloudFront for instance) + what if the operation contains the word `Output` (Let's imagine a `CommandOutputResponse` result..)

I'd rather rely on the operation's name instead `$this->getName().'Request'`